lake/index/expr.go

```diff
@@ -25,6 +25,9 @@ type result struct {
 // - Equals comparisons chained with 'and' or 'or' statements.
 // - Leaf BinaryExprs with the LHS of *dag.Path and RHS of *dag.Literal.
 func compileExpr(node dag.Expr) expr {
+	if node == nil {
+		return nil
+	}
```

Comment:
You don't need this since the type assertion on the next line will fail if `node` is nil.
```suggestion
```

Reply:
Did not know that, thanks.